plotter/labels_test.go

```diff
@@ -5,12 +5,119 @@
 package plotter_test
 
 import (
+	"image/color"
 	"testing"
 
+	"gonum.org/v1/plot"
 	"gonum.org/v1/plot/cmpimg"
+	"gonum.org/v1/plot/plotter"
+	"gonum.org/v1/plot/vg"
+	"gonum.org/v1/plot/vg/draw"
 )
 
 func TestLabels(t *testing.T) {
 	cmpimg.CheckPlot(ExampleLabels, t, "labels.png")
 	cmpimg.CheckPlot(ExampleLabels_inCanvasCoordinates, t, "labels_cnv_coords.png")
 }
+
+// TestLabelsWithGlyphBoxes tests the position of the glyphbox around
+// a block of text, checking whether we correctly take into account
+// the descent+ascent of a glyph.
+func TestLabelsWithGlyphBoxes(t *testing.T) {
+	cmpimg.CheckPlot(
+		func() {
+			const fontSize = 24
+
+			p, err := plot.New()
+			if err != nil {
+				t.Fatalf("could not create plot: %+v", err)
+			}
+
+			p.Title.Text = "Labels"
+			p.X.Min = -1
+			p.X.Max = +1
+			p.Y.Min = -1
+			p.Y.Max = +1
+
+			labels, err := plotter.NewLabels(plotter.XYLabels{
+				XYs: []plotter.XY{
+					{X: -0.8 + 0.00, Y: -0.5}, // Aq + y-align bottom
+					{X: -0.6 + 0.02, Y: -0.5}, // Aq + y-align center
+					{X: -0.4 + 0.04, Y: -0.5}, // Aq + y-align top
+
+					{X: -0.8 + 0.00, Y: +0.5}, // ditto for Aq\nAq
+					{X: -0.6 + 0.02, Y: +0.5},
+					{X: -0.4 + 0.04, Y: +0.5},
+
+					{X: +0.0 + 0.00, Y: +0}, // ditto for Bg\nBg\nBg
+					{X: +0.2 + 0.02, Y: +0},
+					{X: +0.4 + 0.04, Y: +0},
```

Comment:
Can we have above this `NewLabels` call
```
const (
    left   = 0.00
    middle = 0.02
    right  = 0.04
)
```
and use those in place of the literals here and in the following call to `NewLabels`.